aospy/utils/times.py

```diff
@@ -497,3 +497,40 @@ def assert_matching_time_coord(arr1, arr2):
                'arr1: {}\narr2: {}')
     if not (arr1[TIME_STR].identical(arr2[TIME_STR])):
         raise ValueError(message.format(arr1[TIME_STR], arr2[TIME_STR]))
+
+
+def ensure_time_as_dim(ds):
+    """Ensures that time is an indexable dimension on relevant quantites
+
+    In xarray, scalar coordinates cannot be indexed.  We rely
+    on indexing in the time dimension throughout the code; therefore
+    we need this helper method to (if needed) convert a scalar time coordinate
+    to a dimension.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Dataset with a time coordinate
+
+    Returns
+    -------
+    Dataset
+    """
+    TIME_STR = internal_names.TIME_STR
+    GRID_ATTRS_NO_TIMES = internal_names.GRID_ATTRS.copy()
+    del GRID_ATTRS_NO_TIMES[TIME_STR]
+    del GRID_ATTRS_NO_TIMES[internal_names.TIME_BOUNDS_STR]
+    del GRID_ATTRS_NO_TIMES[internal_names.TIME_WEIGHTS_STR]
+
+    if TIME_STR not in ds.dims:
+        _time = [ds[TIME_STR].values.item()]
```

Comment:
Also should we make a general utility method to convert a scalar coordinate into a dimensional one?  I don't know if we'll ever use it anywhere else, but it might make this method a little more readable.
```python

def convert_scalar_coord_to_dim(ds, scalar_coord):
    """Create a DataArray with the given scalar coordinate as a dimension

    Parameters
    ----------
    ds : Dataset
        Dataset containing the scalar coordinate
    scalar_coord : str
        Name of the scalar coordinate

    Returns
    -------
    DataArray
    """
    data = [ds[scalar_coord].values.item()]
    da = xr.DataArray(data, coords=[data], dims=[scalar_coord])
    da.attrs = ds[scalar_coord].attrs
    return da
```